### ISSUE TITLE
Question: it posible build webpack with style-loader

### ISSUE THREAD

**Issue by Bogdaan** [iam4x/isomorphic-flux-boilerplate#162]
Hi, i use webpack style-loader just for optimize long names like "mycomponent-element-element--state-longname", and it imposible use it with this example?

My config

      {
        test: /(\.scss|\.css)$/,
        loader: ExtractTextPlugin.extract('style',
          'css?sourceMap&modules&importLoaders=1&localIdentName=[local]___[hash:base64:5]|postcss!autoprefixer-loader!sass?sourceMap')
      }

In jsx i use requre for classnames map:

    import React, { Component, PropTypes } from 'react';

    import style from './style'

    class Stest extends Component {
      render() {
        return <div className={style['stest']}>stest text</div>
      }
    }}

I must choose: 
or server-side rendering with "global css" and plaintext styles 
or power of webpack with short class names but without server-side rendering.

But I wanted to use both approaches. Anyone else encountered this problem?

**Comment by iam4x** [comment]
Hello @Bogdaan I haven't looked into css-modules yet, I can't help you with that sorry.

**Comment by Bogdaan** [comment]
A possible solution : require hook ( https://github.com/sullenor/universal-demo )

**Comment by Bogdaan** [comment]
I think css-modules is better for application architecture (and for boilerplate). In "real application" we work with hundreds of components. And folder/styles structure must contain full component data (jsx / css / fonts / images / icons ), because the atomic components is easier to maintain.

**Comment by dmitry** [comment]
@Bogdaan have you seen https://github.com/gajus/react-css-modules?

**Comment by Bogdaan** [comment]
@dmitry no, it posible merge to css bundles in one with react-css-modules?
I meen:
1. shared icons (large bundle)
2. component scss (requre icons from bundle 1)

**Comment by iam4x** [comment]
@Bogdaan look at https://github.com/iam4x/isomorphic-flux-boilerplate/pull/167 (thanks to @MaxSvargal) he's using radiumcss. Does it helps?

**Comment by Bogdaan** [comment]
@iam4x i not familiar with radiumcss (this PR large :))
But, i alredy switch my project to simplest decorator like https://github.com/kriasoft/react-starter-kit/blob/master/src/decorators/withStyles.js
with some changes - it solve all problems (shared styles + always critical css + short class names + unique classnames + SSR and more)

**Comment by MaxSvargal** [comment]
@Bogdaan absolutely. Only pure javascript.
I publish complete example with inline css of this boilerplate later.
